src/menu-parser/parsers/compassgroup.ts

```diff
@@ -0,0 +1,80 @@
+import * as moment from 'moment';
+import { JSDOM } from 'jsdom';
+
+import { Parser } from '../index';
+import {
+  text,
+  Property,
+  propertyRegex,
+  createPropertyNormalizer
+} from '../utils';
+
+const dateRegExp = /(\d+).(\d+)./g;
+
+const parseToDate = (possiblyDate: string): string | null => {
+  const match = dateRegExp.exec(possiblyDate);
+  return match
+    ? moment(match[0], 'DD.M').format('YYYY-MM-DD')
+    : null;
+};
+const propertyMap = {
+  PÄ: Property.CONTAINS_ALLERGENS,
+  VS: Property.CONTAINS_ALLERGENS,
+  G: Property.GLUTEN_FREE,
+  V: Property.VEGAN,
+  L: Property.LACTOSE_FREE,
+  M: Property.MILK_FREE,
+  VL: Property.LOW_IN_LACTOSE
+};
+
+const normalizeProperties = createPropertyNormalizer(propertyMap);
+
+interface Course {
+  title: string;
+  properties: Array<Property>;
+};
+
+const parseToCourse = (possiblyCourse: string): Course | null => {
+  if (possiblyCourse.trim() === '') {
+    return null;
+  }
+  const [rawTitle, rawProperties] = possiblyCourse.split(' (');
+  return {
+    title: rawTitle.trim(),
+    properties: normalizeProperties(rawProperties.match(propertyRegex))
```

Comment:
i first tested the parser with [a wrong URL](http://compassgroupfi.episerverhosting.com/Ravintolat/Helsinki/Ravintola-Luova/) and it failed here because `rawProperties` was undefined for a certain `<p>` element which didn't contain ` (`.

on the correct menu page it worked fine, but maybe you could still check if `rawProperties` is falsy before calling `.match()` on it for robustness?

Reply:
Good point! Parser now has check for possibly undefined title and properties.

I also added bettern pattern matching for  compassgroup urls.

Reply:
awesome, thanks! looking good now :)